static/app/components/compactSelect/types.tsx

```diff
@@ -1,6 +1,14 @@
 import {SelectValue} from 'sentry/types';
 
-export interface SelectOption<Value extends React.Key> extends SelectValue<Value> {}
+export interface SelectOption<Value extends React.Key> extends SelectValue<Value> {
+  /**
+   * Whether to hide the checkbox/checkmark. Available for backward compatibility only.
+   * If true, an alternative selection state indicator must be present.
+   *
+   * @deprecated
+   */
+  hideCheck?: boolean;
+}
```

Comment:
It is now possible to hide the leading checkbox. This feature was added specifically for page filters, which use a hybrid selection pattern.
![image](https://user-images.githubusercontent.com/44172267/228975878-63a395fd-34ce-43c7-ae28-8b3a9d01e9b5.png)

The prop for this, `hideCheck`, is marked as deprecated to discourage further use. (If there's a better, more tailored way to restrict usage, pls tell me!).

Reply:
Hmm, it's not ideal to say it's deprecated, as that implies we want to try to remove it.

Reply:
Yeah I agree. Not sure what else to do though. @deprecation was a decent choice because VSCode will add a strikethrough over the name, so at least it'll grab people's attention when they try to use it.

I could also just add a warning in all caps in the prop description, but there's a much lower chance of people seeing it.

Reply:
Lol I tagged an org called Deprecation trying to write `@deprecation`